EncoreTickets.SDK.Tests/Helpers/ApiServiceMockers/MockersForApiService.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using EncoreTickets.SDK.Api.Context;
+using EncoreTickets.SDK.Api.Helpers.ApiRestClientBuilder;
+using EncoreTickets.SDK.Utilities.Common.RestClientWrapper;
+using Moq;
+using RestSharp;
+
+namespace EncoreTickets.SDK.Tests.Helpers.ApiServiceMockers
+{
+    internal class MockersForApiService
+    {
+        public Mock<ApiRestClientBuilder> RestClientBuilderMock;
+
+        public Mock<RestClientWrapper> RestClientWrapperMock;
+
+        public MockersForApiService()
+        {
+            RestClientWrapperMock = GetRestClientWrapperMock();
+            RestClientBuilderMock = GetApiRestClientBuilderMock(RestClientWrapperMock);
+        }
+
+        public void SetupAnyExecution<T>()
+            where T : class, new()
+        {
+            RestClientWrapperMock
+                .Setup(x => x.Execute<T>(It.IsAny<IRestClient>(), It.IsAny<IRestRequest>()))
+                .Returns((IRestClient client, IRestRequest request) => new RestResponse<T>());
+        }
+
+        public void SetupSuccessfulExecution<T>(string responseContent)
+            where T : class, new()
+        {
+            RestClientWrapperMock
+                .Setup(x => x.Execute<T>(It.IsAny<IRestClient>(), It.IsAny<IRestRequest>()))
+                .Returns((IRestClient client, IRestRequest request) =>
+                    RestResponseFactory.GetSuccessJsonResponse<T>(client, request, responseContent));
+        }
+
+        public void SetupFailedExecution<T>(string responseContent, HttpStatusCode code)
+            where T : class, new()
+        {
+            RestClientWrapperMock
+                .Setup(x => x.Execute<T>(It.IsAny<IRestClient>(), It.IsAny<IRestRequest>()))
+                .Returns((IRestClient client, IRestRequest request) =>
+                    RestResponseFactory.GetFailedJsonResponse<T>(client, request, responseContent, code));
+        }
+
+        public void VerifyExecution<T>(
+            string baseUrl,
+            string resource,
+            Method method,
+            Dictionary<string, object> expectedQueryParameters = null,
+            string bodyInJson = null)
```

Comment:
It looks like you are currently using bodyInJson more often than expectedQueryParameters and I think there are generally more requests in the API that have body but not query parameters than those that have it the other way around (please correct me if I'm wrong). In that case it might make sense to swap those parameters